erc20-load-test-tool/script/run-dapp.py

```diff
@@ -0,0 +1,238 @@
+def run_dapp(priv_key, contract_address, port):
+    print("start "+str(port))
+    if(contract_address):
+        os.environ["TOKEN_CONTRACT_ADDRESS"] = contract_address
+
+    mvn = "cd .. ; mvn spring-boot:run -Dspring-boot.run.arguments='--server.port=" + \
+        str(port) + " --token.private-key=" + priv_key + "'"
+
+    p = subprocess.Popen(mvn, shell=True, stdout=subprocess.PIPE)
+    stdout, stderr = p.communicate()
+    print(stderr)
+    if(not p.returncode):
+        print("Dapp with port {} completed with status code {}".format((port), p.returncode)) # is 0 if success
+    
+    return (not p.returncode)
+
+
+# extract kv from list and adds them to dictionary
+def list_to_kv(inp_list, inp_dict):
+    for obj in inp_list:
+        obj_kv = obj.split("=")
+        inp_dict[obj_kv[0]] = inp_dict.get(obj_kv[0],0) + int(obj_kv[1])
+
+
+# reads reports of all runs and write to aggregate-report.json
+def aggregate_report(instance):
+    port= 8080
+    filename = "../output/result/report-"
+    aggregate_throughput = 0
+    aggregate_latency = 0
+    aggregate_tx = 0
+    aggregate_loadfactor = 0
+    aggregate_txStatus = {}
+    aggregate_txErrors = {}
+    aggregate_receiptStatus = {}
+    aggregate_receiptErrors = {}
+    for i in range(1, instance+1):
+        with open(filename+str(port+i)+'.json', 'r') as f:
+            data = json.load(f)
+            aggregate_tx += data['txTotal']
+            aggregate_throughput += data['averageThroughput']
+            aggregate_latency += data['averageLatency']
+            aggregate_loadfactor += data['loadFactor']
+
+            txStatus_list = data["txStatus"].split(",")
+            list_to_kv(txStatus_list, aggregate_txStatus)
+            
+            if(data["txErrors"]):
+                txErrors_list = data["txErrors"].split(",")
+                list_to_kv(txErrors_list, aggregate_txErrors)
+
+            if(data["receiptStatus"]):
+                receiptStatus_list = data["receiptStatus"].split(",")
+                list_to_kv(receiptStatus_list, aggregate_receiptStatus)
+            
+            if(data["receiptErrors"]):
+                receiptErrors_list = data["receiptErrors"].split(",")
+                list_to_kv(receiptErrors_list, aggregate_receiptErrors)
+                
+    json_obj = {}
+    json_obj['aggregate_tx'] = aggregate_tx
+    json_obj['aggregate_throughput'] = aggregate_throughput
+    json_obj['aggregate_latency'] = int(aggregate_latency/instance)
+    json_obj['aggregate_loadfactor'] = aggregate_loadfactor
+    json_obj['aggregate_txStatus'] = aggregate_txStatus
+    json_obj['aggregate_txErrors'] = aggregate_txErrors
+    json_obj['aggregate_receiptStatus'] = aggregate_receiptStatus
+    json_obj['aggregate_receiptErrors'] = aggregate_receiptErrors
+
+    filename = "../output/result/aggregate-report.json"
+    with open(filename, 'w') as f:
+        json.dump(json_obj, f, indent=4)
+
+
+def main():
+    instance = int(os.getenv('DAPP_INSTANCES', 1))
```

Comment:
I'd suggest you get all the relevant parameters upfront for better readability and maintainability.
```python
  host = os.environ['WEB3J_ETHCLIENT_HOST']
  port = os.getenv('WEB3J_ETHCLIENT_PORT', 8585)
  protocol = os.getenv('WEB3J_ETHCLIENT_PROTOCOL', "http")
  dapp_count = int(os.getenv('DAPP_INSTANCES', 1))
  share_contract = os.getenv('SHARE_CONTRACT', 'False') in ('True', 'true')
  ethrpc_url = "{0}://{1}:{2}/".format(protocol, host, port)

  if share_contract:
    assert dapp_count > 1, "At least 2 instances should run to share contract."
```